packages/backend/src/models/SearchModel/utils/search.ts

```diff
@@ -0,0 +1,43 @@
+import { CompiledField, CompiledTable } from '@lightdash/common';
+import { Knex } from 'knex';
+import { compact, escapeRegExp } from 'lodash';
+
+export function getFullTextSearchRankCalcSql(
+    database: Knex,
+    tableName: string,
+    searchVectorColumnName: string,
+    query: string,
+) {
+    return database.raw(
+        `ROUND(
+            ts_rank_cd(
+                :searchVectorColumn:,
+                websearch_to_tsquery('lightdash_english_config', :query),
+                32
+            )::numeric,
+            6
+        )::float`,
+        {
+            searchVectorColumn: `${tableName}.${searchVectorColumnName}`,
+            query,
+        },
+    );
+}
+
+export function getRegexFromUserQuery(query: string) {
+    const sanitizedQuery = escapeRegExp(query);
+    const splitQuery = compact(Array.from(new Set(sanitizedQuery.split(' '))));
+
+    return new RegExp(splitQuery.join('|'), 'ig');
```

Comment:
is the order important?

for example:

if field name is:

dodge
and you search for dodge
will `oddge` have the same result?

Reply:
This regex is matching full words, so switching letters around won't give out the same results